cli/medperf/decorators.py

```diff
@@ -27,3 +31,90 @@ def wrapper(*args, **kwargs):
             pretty_error(str(e))
 
     return wrapper
+
+
+def configurable(with_args: bool = True) -> Callable:
+    """Decorator that adds common configuration options to a typer command
+
+    Args:
+        func (Callable): function to be decorated
+        with_args (bool, optional): Wether to expect arguments or just flags. Defaults to True.
+
+    Returns:
+        Callable: decorated function
+    """
+
+    def decorator(func):
+        atype = str
+        cleanup_opts = "--cleanup/--no-cleanup"
+        default_val = None
```

Comment:
That's true. This was done this way so that we could differentiate assigned vs unassigned config values. I had a thought about keeping configuration profiles slim by only storing the difference from the default configuration, but it might be better to have defaults and always store all configuration variables for each profile. I believe Micah also said that would be more intuitive.

I'll make the required change

Reply:
Hmm will have to look more into this, because providing the default values makes the command think we specified all parameters, which might lead to undesired behavior. For example, running `medperf profile set` will set all parameters to the default values. Additionally, running any medperf command will think we passed all default values as inline cli arguments, invalidating any profile configuration. We can do some workarounds, but in those cases we the `--help` text will have to say None

Reply:
I see your point. I think we should move on with this PR although this "default=None" issue can still be a concern.
A workaround can be to have the `configurable` decorator load the profile and sets the defaults of that specific profile that is being used. for the `main` command where the decorator runs before config initialization, we can move the config initialization logic to happen in the decorator. i.e. have something like @configurable(init_config=True) for the `main` command.

However, this really complicates stuff and adds a lot of logic and an extra argument to the decorator we have, which is already complex.
I think we can move on and merge this PR and create an issue to resolve this later, perhaps to look more into how this type of stuff is handled. So basically we create an issue to resolve later that addresses two confusions:
1) User confusion when they type` --help` and see that the defaults are `None`
2) developer confusion about the `kwargs` in the decorator.

Reply:
Agreed! It's best to move forward and have those concerns as issues so that we can move forward